R/utils.R

```diff
@@ -5,4 +5,122 @@ get_file_type <- function(tracks){
   # character vector:
   #  - .bam$
   #  - .bw$
-}
\ No newline at end of file
+}
+
+
+#check that seqlevelsStyle of x and y match
+#' @noRd
+seqlevels_match <- function(x,y) {
+  
+	if(seqlevelsStyle(x) == seqlevelsStyle(y)){
+		return(TRUE)
+	} else{
+		return(FALSE)		
+	}
+}
+
+
+##this is crude and could be improved
+#check that all the seqlevelsStyles found in X are also found in y
+#' @noRd
+seqlevels_style <- function(x,y) {
+  x_styles <- lapply(seqlevels(x), function(a) seqlevelsStyle(a))
+  y_styles <- lapply(seqlevels(y), function(a) seqlevelsStyle(a))
+  if(!all(x_styles %in% y_styles)){
+    return(FALSE)
+  }else {
+    return(TRUE)
+  }
+}
+
+#check that the seqnames in x are valid - relying on built in error for invalid seqlevelsStyle
+#' @noRd
+seqlevels_valid <- function(x) {
+  out <- lapply(seqlevels(x), function(y) {
+    check <- try(seqlevelsStyle(y), silent = TRUE)
+    if(class(check) == "try-error"){
+      return(FALSE)
+    } else{
+      return(TRUE)
+    }
+    return(check)
+  })
+  
+  suppressWarnings(if(!all(out)){
+    return(FALSE)
+  }else{
+    return(TRUE)
+  })
+  
+}
+
+
+#check if seqlevels of x are a subset of seqlevels in y
+#' @noRd
+seqlevels_all_within <- function(x, y) {
+  if(all(seqlevels(x) %in% seqlevels(y))) {
+    return(TRUE)
+  } else {
+    return(FALSE)
+  }
+}
+
+#return any levels in x that are missing in y 
+#' @noRd
+seqlevels_missing <- function(x,y) {
+  missing_levels <- seqlevels(x)[!seqlevels(x) %in% seqlevels(y)]  
+  return(missing_levels)
+}
+
+#return any levels in x that are shared in y 
+#' @noRd
+seqlevels_shared <- function(x,y) {
+  if(any(seqlevels(x) %in% seqlevels(y))) {
+    shared_levels <- seqlevels(x)[seqlevels(x) %in% seqlevels(y)]  
+    return(shared_levels)
+  } else {
+    return(FALSE)
```

Comment:
If getting that `FALSE` return is important later, you could refactor a bit to add a second function: 

```
has_shared_seqlevels <- function(x,y){
  any(seqlevels(x) %in% seqlevels(y))
}
```

And this will give `TRUE`/`FALSE` if there are any shared or not. This way you have type stability for both functions (as much as returning `NULL` vs a `character()` is type stable...).